File: code_challenge_solution/javascript/208 Typing speed calculator.js
let startTime, endTime;

function startTimer() {
  startTime = new Date();
}

function endTimer() {
  endTime = new Date();
}

function calculateTypingSpeed(text, time) {
  const words = text.trim().split(" ");
  const wordCount = words.length;
  const totalTimeInSeconds = (time - startTime) / 1000;
  const wordsPerMinute = Math.round((wordCount / totalTimeInSeconds) * 60);

  return wordsPerMinute;
}

// Example usage
const text = "The quick brown fox jumps over the lazy dog";
startTimer();
// User types the sentence...
endTimer();

const speed = calculateTypingSpeed(text, endTime);
console.log("Typing speed:", speed, "words per minute");

File: code_challenge_solution/python/208 Typing speed calculator.py
import time

def calculate_typing_speed():
    start_time = time.time()
    text = input("Type the following text:\n")
    end_time = time.time()

    num_chars = len(text)
    num_words = len(text.split())
    time_taken = end_time - start_time

    speed_cpm = (num_chars / time_taken) * 60
    speed_wpm = (num_words / time_taken) * 60

    print("Typing speed:")
    print("Characters per minute:", speed_cpm)
    print("Words per minute:", speed_wpm)

if __name__ == "__main__":
    calculate_typing_speed()